src/main/java/org/prebid/server/bidder/bidmachine/BidmachineBidder.java

```diff
@@ -0,0 +1,235 @@
+                    return BidType.video;
+                } else {
+                    return BidType.banner;
+                }
+            }
+        }
+
+        errors.add(BidderError.badServerResponse(String.format("Failed to find impression with id: %s", impId)));
+        return null;
+    }
+
+    private BidRequest createSingleRequest(Imp imp, ExtImpPrebid extPrebid, BidRequest request) {
+        return request.toBuilder()
+                .imp(Collections.singletonList(modifyImp(imp, extPrebid)))
+                .build();
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpPrebid extPrebid) {
+        if (extPrebid != null && extPrebid.getIsRewardedInventory() == 1) {
+            final Banner banner = imp.getBanner();
+            final List<Integer> bannerBattr = banner == null ? null : banner.getBattr();
+            final Imp.ImpBuilder impBuilder = imp.toBuilder();
+            if (bannerBattr != null && !hasRewardedBattr(bannerBattr)) {
+                bannerBattr.add(16);
+                impBuilder.banner(imp.getBanner().toBuilder().battr(bannerBattr).build());
+            }
+            final Video video = imp.getVideo();
+            final List<Integer> videoBattr = video == null ? null : video.getBattr();
+
+            if (videoBattr != null && !hasRewardedBattr(videoBattr)) {
+                videoBattr.add(16);
+                impBuilder.video(imp.getVideo().toBuilder().battr(videoBattr).build());
+            }
+
+            return impBuilder.build();
+        }
+
+        return imp;
+    }
+
+    private boolean hasRewardedBattr(List<Integer> battr) {
+        return battr.contains(16);
+    }
+
+    private ExtImpPrebid parsePrebid(ObjectNode ext) throws JsonProcessingException {
+        return mapper.mapper().treeToValue(ext.get("prebid"), ExtImpPrebid.class);
+    }
+
+    private String buildEndpointUrl(ExtImpBidmachine extImpBidmachine) {
+        return endpointUrl.replace("{{HOST}}", extImpBidmachine.getHost())
+                .replace("{{PATH}}", extImpBidmachine.getPath())
+                .replace("{{SELLER_ID}}", extImpBidmachine.getSellerId());
+    }
+
+    private ExtImpBidmachine parseAndValidateImpExt(Imp imp) {
+        final ExtImpBidmachine extImpBidmachine;
+        try {
+            extImpBidmachine = mapper.mapper().convertValue(imp.getExt(), BIDMACHINE_EXT_TYPE_REFERENCE).getBidder();
+        } catch (JsonParseException e) {
+            throw new PreBidException("Missing bidder ext in impression with id: " + imp.getId());
+        }
+
+        if (StringUtils.isEmpty(extImpBidmachine.getSellerId())) {
+            throw new PreBidException("Invalid/Missing sellerId");
+        }
+
+        if (StringUtils.isBlank(extImpBidmachine.getHost())) {
+            throw new PreBidException("Invalid/Missing host");
+        }
+
+        if (StringUtils.isEmpty(extImpBidmachine.getPath())) {
+            throw new PreBidException("Invalid/Missing path");
+        }
```

Comment:
There is no such validation in go, and actually we don't need it, because we have Json validation schemas